src/lib/service/handlers.ts

```diff
@@ -0,0 +1,108 @@
+import { CreateContext, DeleteContext, DetailContext, ListContext, UpdateContext } from 'lib/context/crudContext';
+import { Operation } from 'lib/context/operation';
+import { errorOrEmpty } from 'lib/helpers';
+import { ServiceImplementation } from 'lib/settings/definitions';
+
+export const createHandlers = <T extends { id: any }, C extends object>(
+    implementation: ServiceImplementation<T, C>
+) => {
+    /**
+     * Fetch resource, throw error when resource missing.
+     * This method is used for handlers working with a single existing resource (get, update, delete)
+     */
+    const getSafe = (context: Pick<DetailContext<T, C>, 'id' | 'context' | 'options'>): Promise<T> =>
+        implementation
+            .detail({ ...context, type: Operation.DETAIL, write: false, safe: true })
+            .then(errorOrEmpty(implementation.createNotFoundError()));
+
+    const detailHandler = (options: any = {}) => async (id: number, context: C) => {
+        const dynamicOptions = await implementation.getOptions(Operation.DETAIL);
+        options = { ...dynamicOptions, ...options, ...(context as object) };
+        const ctx: DetailContext<T, C> = await getSafe({ id, context, options }).then(entity => ({
```

Comment:
Is it necessary to combine await and then in the same function?

If you really want to use the function-style, I'd prefer an IIEF to do this:
```
ctx = await ((entity) => { ...})(await getSafe(...))
```
This should do just fine.

Reply:
Agree it is distasteful. However this is even worse IMHO form the point of readability :crying_cat_face: 

Will refactor.

Reply:
Fixed :heavy_check_mark: